_posts/2019-06-06-NeuralODEs.markdown

```diff
@@ -0,0 +1,388 @@
+     
+     $$y_{n+1} = y_n + hy_n' + h^2 y_n'' + h^3 y_n''' + h^4y_n^{(4)} + O(h^5)$$
+     
+     $$y_{n+2}' = y_n' + 2hy_n'' + 2h^2y_n''' + 4/3 h^3 y_{n}^{(4)}$$
+     
+     $$y_{n+1}' = y_n' + hy_n'' + h^2y_n''' + h^3 y_{n}^{(4)}$$
+     
+     Substituting these in the first equation and imposing the terms in \(h^i\), \(i = 0,1,2,3,4\), to be \(0\), we get the equations
+     
+     $$3 + \alpha - 2\beta = 0$$
+     
+     $$27 + 7\alpha - 15\beta = 0$$
+     
+     $$27 + 5\alpha - 12\beta = 0$$
+     
+     Solving for these, we find \(\alpha = 9\) and \(\beta = 6\). The resulting method reads
+     
+     $$y_{n+3} + 9(y_{n+2} - y_{n+1}) - y_n = 6h(f_{n+2} + f_{n+1})$$
+     
+     The characteristic polynomial is given by
+     
+     $$\rho(z) = z^3 +9z^2 - 9z -1$$
+     
+     One of the roots of this polynomial satisfies \(|z|>1\) and this implies that the method is not zero-stable.
+     </p>
+     <p>
+     Exercise 12.12:
+     By definition, the truncation error is given by
+     
+     $$h\tau_n = y_{n+1} + b y_{n-1} +a y_{n-2} -h y_{n}'$$
+     
+     Taylor-expanding, we have that
+     
+     $$y_{n+1} = y_n + hy_n' + 1/2h^2 y_n'' + O(h^3)$$
+     
+     $$y_{n-1} = y_n - hy_n' + 1/2h^2 y_n'' + O(h^3)$$
+     
+     $$y_{n-2} = y_n - 2hy_n' + 2h^2 y_n'' +  O(h^3)$$
+     
+     Substituting these in the first equation and solving for the terms in \(h^i\), \(i = 0,1\), to be \(0\), we get \(a=1\) and \(b=-2\). In particular
+     
+     $$\tau_n = 3/2h + O(h^2)$$
+     
+     and thus the method has order of accuracy \(1\).
+     The resulting method reads
+     
+     $$y_{n+1} -2 y_{n-1} + y_{n-2} = h f_{n}$$
+     
+     The characteristic polynomial is given by
+     
+     $$\rho(z) = z^3 -2z -1$$
+     
+     One of the roots of this polynomial satisfies \(|z|>1\) and this implies that the method is not zero-stable.
+     </p>
+     <p>
+     Exercise 12.19:
+     The first equation can be found by substituting \(f(t,y) = \lambda y\) in equation (12.51) in the book and by solving for \(k_1,k_2\) (it is a \(2\times 2\) linear system). Substituting the values of \(A\) and \(b\) from the Butcher tableau in this formula and in the one right before equation (12.51) in the book, and simplifying, we get the formula for \(R(\lambda h)\). Finally, \(p\) and \(q\) are given by \(p,q=-3\pm i \sqrt{3}\). One can see that this implies \(|R(z)|<1\) if \(Re(z) <0\) and thus the method is A-stable.
+     </p>
+     </details>
+
+<br />
+
+# 3 ResNets
+  **Motivation**: The introduction of Residual Networks (ResNets) made possible to train very deep networks. In this section we study some residual architectures variants and their properties. We then look into how ResNets approximates ODEs and how this interpretation can motivate neural net architectures and new training approaches.  This is important in order to understand the basic models underlying Neural ODEs and gain some insights on their connection to numerical solutions of ODEs.
```

Comment:
"made *it* possiblde"

Reply:
--> "In this section, we study residual architectures and their properties."

Reply:
"gain some insights into their connection"

Reply:
✅

Reply:
- "look into how ResNets approximate ODEs" (s/approximates/approximate)
- "insights on" --> "insights into"